nextjs-wagmi-v0.12.x/src/pages/_app.jsx

```diff
@@ -0,0 +1,60 @@
+import {
+  EthereumClient,
+  w3mConnectors,
+  w3mProvider,
+} from "@web3modal/ethereum";
+import { Web3Modal } from "@web3modal/react";
+import { useEffect, useState } from "react";
+import { configureChains, createClient, WagmiConfig } from "wagmi";
+import {
+  arbitrum,
+  avalanche,
+  bsc,
+  fantom,
+  gnosis,
+  mainnet,
+  optimism,
+  polygon,
+} from "wagmi/chains";
+import "../styles.css";
+
+// 1. Get projectID at https://cloud.walletconnect.com
+if (!process.env.NEXT_PUBLIC_PROJECT_ID) {
+  throw new Error("You need to provide NEXT_PUBLIC_PROJECT_ID env variable");
+}
+const projectId = process.env.NEXT_PUBLIC_PROJECT_ID;
+
+// 2. Configure wagmi client
+const chains = [
+  mainnet,
+  polygon,
+  avalanche,
+  arbitrum,
+  bsc,
+  optimism,
+  gnosis,
+  fantom,
+];
+
+const { provider } = configureChains(chains, [w3mProvider({ projectId })]);
+const wagmiClient = createClient({
+  autoConnect: true,
+  connectors: w3mConnectors({ version: 1, chains, projectId }),
```

Comment:
This still targets WalletConnect v1 with this parameter, I believe?

Reply:
correct, should be 2